cirq/pasqal/pasqal_qubits.py

```diff
@@ -192,3 +193,194 @@ def __rsub__(self, other: Tuple[int, int, int]) -> 'ThreeDGridQubit':
+
+        >>> cirq.pasqal.ThreeDQubit(2.5, 3, 4.7) + (3, 1.2, 6)
+        pasqal.ThreeDQubit(5.5, 4.2, 10.7)
+
+        >>> cirq.pasqal.ThreeDQubit(2.4, 3.1, 4) - (1, 2, 2.1)
+        pasqal.ThreeDQubit(1.4, 1.1, 1.9)
+    """
+
+    def __init__(self, x: float, y: float, z: float):
+        self.x = x
+        self.y = y
+        self.z = z
+
+    def _comparison_key(self):
+        return round(self.z, 9), round(self.y, 9), round(self.x, 9)
+
+    @property
+    def dimension(self) -> int:
+        return 2
+
+    def distance(self, other: cirq.ops.Qid) -> float:
+        """Returns the distance between two qubits in 3d."""
+        if not isinstance(other, ThreeDQubit):
+            raise TypeError(
+                "Can compute distance to another ThreeDQubit, but {}".format(
+                    other))
+        return sqrt((self.x - other.x)**2 + (self.y - other.y)**2 +
+                    (self.z - other.z)**2)
+
+    @staticmethod
+    def cube(diameter: int, x0: float = 0, y0: float = 0,
+             z0: float = 0) -> List['ThreeDQubit']:
+        """Returns a cube of ThreeDQubits.
+
+        Args:
+            diameter: Length of a side of the square
+            x0: x-coordinate of the first qubit
+            y0: y-coordinate of the first qubit
+            z0: z-coordinate of the first qubit
+
+        Returns:
+            A list of ThreeDQubits filling in a square grid
+        """
+        return ThreeDQubit.parallelep(diameter,
+                                      diameter,
+                                      diameter,
+                                      x0=x0,
+                                      y0=y0,
+                                      z0=z0)
+
+    @staticmethod
+    def parallelep(rows: int,
+                   cols: int,
+                   lays: int,
+                   x0: float = 0,
+                   y0: float = 0,
+                   z0: float = 0) -> List['ThreeDQubit']:
+        """Returns a parallelepiped of ThreeDQubits.
+
+        Args:
+            rows: Number of rows in the rectangle
+            cols: Number of columns in the rectangle
+            x0: x-coordinate of the first qubit
+            y0: y-coordinate of the first qubit
+            z0: z-coordinate of the first qubit
+
+        Returns:
+            A list of ThreeDQubits filling in a 3D grid
+        """
+        return [
+            ThreeDQubit(x0 + x, y0 + y, z0 + z) for z in range(lays)
+            for y in range(cols) for x in range(rows)
+        ]
+
+    def __repr__(self):
+        return 'pasqal.ThreeDQubit({}, {}, {})'.format(self.x, self.y, self.z)
+
+    def __str__(self):
+        return '({}, {}, {})'.format(self.x, self.y, self.z)
+
+    def _json_dict_(self):
+        return cirq.protocols.obj_to_dict_helper(self, ['x', 'y', 'z'])
+
+    def __add__(self, other: Tuple[float, float, float]) -> 'ThreeDQubit':
+        if not (isinstance(other, tuple) and len(other) == 3 and
```

Comment:
one last question: `GridQubit` does have the functionality to add/substract `GridQubit`s together in here: 

https://github.com/quantumlib/Cirq/blob/adc8964697feb9f97474ab13d60c36a6f27f3bd4/cirq/devices/grid_qubit.py#L76-L82

Do you mind adding that with tests to be consistent?

Reply:
Actually, I've been discussing with the rest of the team and we reached the conclusion that these dunder methods for addition and subtraction of qubits don't really make sense for our case. 
They were first added because the `ThreeDGridQubit` class tried to mimic `GridQubit` and were just left there for no particular reason. As the `ThreeDGridQubit` class turned into `ThreeDQubit`, it made increasingly less sense to have them, so I am now advocating for their removal. However, before I do so, I would like to know,  do you have any objection or counter-argument to this?

Reply:
No counter argument / objection on my end - I prefer not having it at all then having it half-way implemented!